spock-core/src/main/java/org/spockframework/runtime/extension/builtin/TagExtension.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.spockframework.runtime.extension.builtin;
+
+import org.spockframework.runtime.extension.IAnnotationDrivenExtension;
+import org.spockframework.runtime.model.FeatureInfo;
+import org.spockframework.runtime.model.SpecElementInfo;
+import org.spockframework.runtime.model.SpecInfo;
+import spock.lang.Tag;
+
+/**
+ * Processes {@link Tag @Tag} annotations on specifications and features, converting them to
+ * {@link org.spockframework.runtime.model.Tag Spock tags} which can potentially be used by reporting extensions.
+ * More importantly, it also makes the tag information available to JUnit 5 as
+ * {@link org.junit.platform.engine.TestTag platform tags} in order to enable filtering, e.g. from build tools like
+ * Maven or Gradle.
+ *
+ * @see <a href="https://junit.org/junit5/docs/current/user-guide/#running-tests-tag-expressions">
+ *   JUnit platform tag expression syntax</a>
+ * @author Alexander Kriegisch
+ * @since 2.2
+ */
+public class TagExtension implements IAnnotationDrivenExtension<Tag> {
+  public static final String TAG_EXTENSION_KEY = "TagExtension";
```

Comment:
🤔 as this key is also surfaced to reporting, it should probably be more in line with the other keys, e.g. `testTag`.

Reply:
At first, locally it was "junit-platform-tag". Then I wanted to avoid the JUnit reference and changed it to "test-tag". But then I thought that that would be an obvious name for anyone experimenting with Spock tags and lead to a key name clash. It sounds like a name I would use in an automated test. Therefore I simply went with the extension's simple class name, making a clash much less likely.

Reply:
This kind of conflicts with what I had in mind when I said it should be done via extensions, so that other extensions could also add tags. Now the have to use this `TagExtension` key although they are not the `TagExtension`. Furthermore, this also kind of conflicts a bit with the general purpose tag you mentioned above.

> But then I thought that that would be an obvious name for anyone experimenting with Spock tags and lead to a key name clash. It sounds like a name I would use in an automated test.

How would that be triggered, they'd have to write their own extension to access `Tag`?

Reply:
We seem to misunderstand each other here: For me `@Tag` is a general-purpose tag (from user perspective) and `TagExtension` makes sure the annotation values are mapped to general-purpose Spock tags with the same general-purpose key. There is no connection to JUnit whatsoever up to this point. Neither did I say that other special-purpose annotations and extensions cannot be built. Where is the problem here?

As an additional benefit, those general-purpose tags can also be automatically mapped to JUnit tags, if the user happens to filter for those. So that would be one purpose in the larger set of general purposes. The `TagExtension` is agnostic of it, which is the beauty of this design.

> How would that be triggered, they'd have to write their own extension to access `Tag`?

No, they would simply have to use Spock tags from another extension (or from an integration test using Spock tags) and select the same key "test-tag". Even without knowing about the existence of `@Tag` or `TagExtension`, there would be a name clash, because the key name is such an obvious choice for someone testing something. Developers do that, don't they? And again, the idea behind `@Tag` is not that is it limited to be a JUnit test tag, but it can be any tag.

Reply:
> as this key is also surfaced to reporting, it should probably be more in line with the other keys, e.g. `testTag`.

In commit 2015f7fa, I renamed the key to "testTag". Whatever, it is just a name, and if you insist, no problem. Whoever as an extension developer randomly happens to use the same key, will have to debug by herself if there is an accidental naming clash.